### PR TITLE
Dennis dev portfolio preview and validation

### PR DESCRIPTION
**Pull request by denniswillie**
1. Add image preview for when user uploads a new image. 
2. Added validation for form using html regex.

### CODE REVIEW

File: src/main/webapp/createPlaceGuide.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            <div id = "imageDiv">
+            <br>
+            <img alt = "image preview" id = "imagePreview">
+            <br>
+            </div>
+            <label for = "deleteImage">Delete previous image</label>
+            <input type = "checkbox" name = "deleteImage" value = "delete" id = "deleteImage">
+            <br>
+            <label for = "description">Description:</label><br>
+            <textarea name = "description" rows = "4" cols = "50" id = "description" pattern = "[A-Za-z0-9\s]+" title = "Enter alphanumeric string!"></textarea>
```
Comment (bori00) [inline src/main/webapp/createPlaceGuide.html]: "Enter alphanumeric string" doesn't really sound user-friendly.

Comment (bori00) [inline src/main/webapp/createPlaceGuide.html]: Shouldn't we set a maximum length for the description?

File: src/main/webapp/createPlaceGuide.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            <li><a href = "createPlaceGuide.html">Create Place Guide</a><li>
+        </ul>
+    </header>
+    <div id="createPlaceGuide">
+        <form method = "POST" id = "createPlaceGuideForm" enctype = "multipart/form-data">
+            <input type="hidden" id="id" name="id">
+            <input type="hidden" id="placeId" name="placeId">
+            <br>
+            <label for = "name">Name:</label><br>
+            <input type = "text" name = "name" id = "name" required pattern = "[A-Za-z0-9\s]+" title = "Enter alphanumeric string!">
```
Comment (bori00) [inline src/main/webapp/createPlaceGuide.html]: Maximum length?

File: src/main/webapp/index.html [hunk 3]
```diff
@@ -13,31 +14,39 @@
...
     </header>
-  <div id="createProfile">
-    <form method = "POST" id = "portfolioForm" enctype = "multipart/form-data">
-        <label for = "name">Name:</label><br>
-        <input type = "text" name = "name" id = "name"><br>
 
-        <label for = "imgKey">Upload image</label>
-        <input type = "file" name = "imgKey" id = "imgKey">
+    <div id="createProfile">
+        <form method = "POST" id = "portfolioForm" enctype = "multipart/form-data">
```
Comment (bori00) [inline src/main/webapp/index.html:22]: Comments for placeGuide form are valid here too.

File: src/main/webapp/createPlaceGuide.html [hunk 4]
```diff
@@ -40,7 +40,7 @@
             <input type = "checkbox" name = "deleteImage" value = "delete" id = "deleteImage">
             <br>
             <label for = "description">Description:</label><br>
-            <textarea name = "description" rows = "4" cols = "50" id = "description" pattern = "[A-Za-z0-9\s]+" title = "Enter alphanumeric string!"></textarea>
+            <textarea name = "description" rows = "4" cols = "50" id = "description" pattern = "[A-Za-z0-9,.\s]+" title = "You can only enter alphabets, numbers, periods, and comas!"></textarea>
```
Comment (bori00) [inline src/main/webapp/createPlaceGuide.html]: The alphabet is a set of letters, in this case, you meant letters, right?

File: src/main/webapp/index.html [hunk 5]
```diff
@@ -36,7 +36,7 @@
             <input type = "checkbox" name = "deleteImg" value = "delete" id = "deleteImg"><br>
 
             <label for = "selfIntroduction">Self Introduction:</label><br>
-            <textarea name = "selfIntroduction" rows = "4" cols = "50" id = "selfIntroduction" pattern = "[A-Za-z0-9\s]+" title = "Enter alphanumeric string!"></textarea><br>
+            <textarea name = "selfIntroduction" rows = "4" cols = "50" id = "selfIntroduction"  maxlength = "200" pattern = "[A-Za-z0-9,.\s]+" title = "You can only enter alphabets, numbers, periods, and comas!"></textarea><br>
```
Comment (bori00) [inline src/main/webapp/index.html:39]: Same.

**Review by bori00 — COMMENTED** [review]
LGTM!

File: src/main/webapp/createPlaceGuide.html (hunk 4 above)
Comment (denniswillie) [reply src/main/webapp/createPlaceGuide.html]: changed.

File: src/main/webapp/index.html (hunk 5 above)
Comment (denniswillie) [reply src/main/webapp/index.html:39]: changed.

File: src/test/java/com/google/sps/PlaceGuideServletTest.java [hunk 6]
```diff
@@ -0,0 +1,220 @@
+// // Copyright 2020 Google LLC
```
Comment (tlheng) [inline src/test/java/com/google/sps/PlaceGuideServletTest.java:1]: comment out everything?

**Review by tlheng — COMMENTED** [review]
This change is too big. It is very hard to know what it does and review. Please break it into multiple smaller change.

**Comment by denniswillie** [comment]
closed due to too many changes in the pull request.

**Pull request closed without merging**